src/main/java/org/tron/common/overlay/discover/table/NodeEntry.java

```diff
@@ -70,7 +70,7 @@ public long getModified() {
   public boolean equals(Object o) {
     boolean ret = false;
 
-    if (o instanceof NodeEntry) {
+    if (o == null || this.getClass() == o.getClass()) {
       NodeEntry e = (NodeEntry) o;
       ret = this.getId().equals(e.getId());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "e" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=853&issues=AWOhC8fGp7N0pg-JwXPB&open=AWOhC8fGp7N0pg-JwXPB)

[ISSUE_KEY]: AWOhC8fGp7N0pg-JwXPB